src/griffe/extensions/dataclasses.py

```diff
@@ -176,13 +176,25 @@ def _set_dataclass_init(class_: Class) -> None:
     class_.set_member("__init__", init)
 
 
+def _del_members_annotated_as_initvar(class_: Class) -> None:
+    # Definitions annotated as InitVar are not class members
+    attributes = cast(
+        list[Attribute],
+        [member for member in class_.members.values() if member.is_attribute]
+    )
```

Comment:
Unfortunately this doesn't work on Python 3.8 (and 3.9 I think), as types are not subscriptable until 3.10. You can simply remove the cast and use a `# type: ignore[...]` comment.

Reply:
Also, it could be preferable to use `attributes = [member for member in class_.members.values() if isinstance(member, Attribute)`. This way you don't need the type ignore comment or the cast, and we also rule out aliases, since they couldn't possibly be annotated with `InitVar` anyway.